File: minesweeper/python/test_minesweeper.py
import unittest
from Minesweeper import Minesweeper

class TestMinesweeper(unittest.TestCase):

    def test_command_create_a_new_game(self):
        minesweeper = Minesweeper()
        minesweeper.new_game()

    def test_query_view_the_board(self):
        minesweeper = Minesweeper()
        minesweeper.new_game()
        board = minesweeper.view_board()
        # by default, board initialises to 10 x 10 tiles
        # unrevealed tiles are marked by spaces ' '
        self.assertEqual(board, [
            [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
            [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
            [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
            [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
            [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
            [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
            [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
            [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
            [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
            [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
        ])

    def test_query_mines_remaining(self):
        minesweeper = Minesweeper()
        minesweeper.new_game()
        self.assertEqual(minesweeper.mines_remaining(), 10)

    def test_command_reveal_position(self):
        minesweeper = Minesweeper()
        minesweeper.new_game()
        minesweeper.reveal({
            'row': 0,
            'col': 0
        })

    def test_command_mass_reveal_position(self):
        minesweeper = Minesweeper()
        minesweeper.new_game()
        minesweeper.reveal({'row': 0,'col': 0})
        minesweeper.mass_reveal({'row': 0,'col': 0})

    def test_command_toggle_mine_marking(self):
        minesweeper = Minesweeper()
        minesweeper.new_game()
        minesweeper.toggle_mine_marking({'row': 0,'col': 0})

    def test_query_game_status(self):
        minesweeper = Minesweeper()
        minesweeper.new_game()
        self.assertEqual(minesweeper.status(), 'Playing')



if __name__ == '__main__':
    unittest.main()

File: minesweeper/dart/flutter/minesweeper_flutter_app/lib/src/domain/test_minesweeper.dart
import 'package:test/test.dart';
import './Minesweeper.dart';

void main() {
  test('Command: create a new game', () {
    Minesweeper minesweeper = Minesweeper();
    minesweeper.new_game();
  });

  test('Query: view the board', () {
    Minesweeper minesweeper = Minesweeper();
    minesweeper.new_game();
    expect(minesweeper.view_board(), [
      [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
    ]);
  });

  test('Query: mines remaining', () {
    Minesweeper minesweeper = Minesweeper();
    minesweeper.new_game();
    expect(minesweeper.mines_remaining(), 10);
  });

  test('Command: reveal position', () {
    Minesweeper minesweeper = Minesweeper();
    minesweeper.new_game();
    minesweeper.reveal({
      'row': 0,
      'col': 0
    });
  });

  test('Command: mass reveal position', () {
    Minesweeper minesweeper = Minesweeper();
    minesweeper.new_game();
    minesweeper.reveal({'row': 0,'col': 0});
    minesweeper.mass_reveal({'row': 0,'col': 0});
  });

  test('Command: toggle mine marking', () {
    Minesweeper minesweeper = Minesweeper();
    minesweeper.new_game();
    minesweeper.toggle_mine_marking({'row': 0,'col': 0});
  });

  test('Query: game status', () {
    Minesweeper minesweeper = Minesweeper();
    minesweeper.new_game();
    expect(minesweeper.status(), 'Playing');
  });

  test('Query: time elapsed', () {
    Minesweeper minesweeper = Minesweeper();
    minesweeper.new_game();
    expect(minesweeper.time_elapsed(), 0);
  });
}
